app/models/proposal.rb

```diff
@@ -75,12 +75,20 @@ def self.recommendations(user)
       .where("author_id != ?", user.id)
       .unsuccessful
       .not_followed_by_user(user)
+      .not_archived
+      .not_supported_by_user(user)
   end
 
   def self.not_followed_by_user(user)
     where.not(id: followed_by_user(user).pluck(:id))
   end
 
+  def self.not_supported_by_user(user)
+    @supported_by_user = user.find_voted_items(votable_type: "Proposal")
```

Comment:
Not sure why we need a class variable `@supported_by_user` instead of just a variable `supported_by_user`

Reply:
Not needed, actually, but it feels kinda weird using local variables.

Reply:
But weird in what way? I mean the scope of the variable (global, class, local, constant, instance...) it

Reply:
I know, I know. I'm just so accustomed to instance variables (because of how common they are) that it's weird for me to use local variables 😅 But don't worry, I'll fix that.

Reply:
But the point of instance variables is to be able to use them in other places, if its not needed then 🤔 , well there's also [memoization](https://www.justinweiss.com/articles/4-simple-memoization-patterns-in-ruby-and-one-gem/) but still this scenario doesn't ask for it. Problem with using something not needed is that it distracts the code reader by making he guess what's going on and where's that instance variable used as well.

Reply:
And I'm talking about it like this is totally wrong or it could lead to bad performance... we could totally let it there forever. But since you seem concerned as well with code consistency & readability (because I see you normally check with rubocop the files you touch) maybe you saw it the same way as me